### PR TITLE
clean boilerplate and use mariaDB

### PR DESCRIPTION
**Pull request by SwannHERRERA**
### Added
- mariaDB config
- endPoint Hello World on '/api'
- inquired

### Changed
- userService use of Repository instead of MongodbRepository
- database/seeds and database/factory the transition from MongoDB to MariaDB was destructive here.
we should probably rething this part of the project later
- config : use of full db description insteed of db_url and split it between env
- seed (by cli) tools

### Removed
- markdown
- Unused Routes and models (Company and JobApplication)
- commander

### CODE REVIEW

**Review by Huriumari — APPROVED** [review]

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+version: '3'
```
Comment (Nouuu) [inline docker-compose.yml:1]: Tu l'utilises ?

File: package.json [hunk 2]
```diff
@@ -4,10 +4,9 @@
   "description": "Boilerplate code to get started with building RESTful API Services",
   "main": "dist/app.js",
   "scripts": {
-    "start": "node dist/app.js",
+    "start": "cross-env NODE_ENV=production node dist/app.js",
```
Comment (Nouuu) [inline package.json:7]: kesako ?

File: package.json [hunk 3]
```diff
@@ -56,7 +53,7 @@
     "helmet": "^4.1.1",
     "jsonwebtoken": "^8.5.1",
     "lodash": "^4.17.20",
-    "mongodb": "^3.6.2",
+    "mysql2": "^2.2.5",
```
Comment (Nouuu) [inline package.json:56]: il existe un driver mariadb ?

File: src/api/entities/User.ts [hunk 4]
```diff
@@ -27,24 +32,29 @@ export class User {
...
+  user: User;
+  token: string;
+}
+
+export interface userCreationProps {
+  firstName: string;
+  lastName: string;
+  email: string;
+  password: string;
+}
```
Comment (Nouuu) [inline src/api/entities/User.ts:60]: Tu pourras me recap ce qu'on fait avec ça ?

File: src/api/routes/user.ts [hunk 5]
```diff
@@ -23,7 +24,7 @@ route.get(
   }
 );
 
-route.get('/current', isAuth, attachUser, (req: Request, res: Response) => {
+route.get('/current', isAuth, attachUser, (req: userRequest, res: Response) => {
```
Comment (Nouuu) [inline src/api/routes/user.ts:27]: ya moyen d'avoir un exemple d'utilisation du checkRole aussi ?

**Review by Nouuu — APPROVED** [review]
GGWP

**Pull request merged**